src/main/java/interface_adapter/edit_assignment/EditAssignmentPresenter.java

```diff
@@ -0,0 +1,31 @@
+package src.main.java.interface_adapter.edit_assignment;
+
+import src.main.java.use_case.edit_assignment.EditAssignmentOutputBoundary;
+import src.main.java.use_case.edit_assignment.EditAssignmentOutputData;
+
+/**
+ * The Presenter for the Edit Assignment Use Case.
+ */
+public class EditAssignmentPresenter implements EditAssignmentOutputBoundary {
+
+    private final EditAssignmentViewModel editAssignmentViewModel;
+
+    public EditAssignmentPresenter(EditAssignmentViewModel editAssignmentViewModel) {
+        this.editAssignmentViewModel = editAssignmentViewModel;
+    }
+
+    @Override
+    public void prepareSuccessView(EditAssignmentOutputData outputData) {
+        // currently there isn't anything to change based on the output data,
+        // since the output data only contains the new score
+        // We still fire the property changed event, but just to let the view know that
+        // it can alert the user that their assignment was changed successfully..
+        editAssignmentViewModel.firePropertyChanged("Assignment Edited");
+
+    }
+
+    @Override
+    public void prepareFailView(String error) {
+        // note: this use case currently can't fail
```

Comment:
what about when the new assignment name is the same as a dif assignment name? does that not need a failview?